### PR TITLE
Fix SRID not being optional for expression in GeometryCast

### PR DESCRIPTION
**Pull request by nickknissen**
The current implementation of GeometryCast requires SRID to present in the expression when setting an model attribute with `DB:raw`  

The given case:
```php
$place = Place::factory()->create(['point' => DB::raw("ST_GeomFromText('POINT(12.38057 55.73406)')")]);
$place->isDirty()
   ```
This will throw the following error because the regex looks for a SRID in the expressionValue
```
  Undefined array key 1

  at src\GeometryCast.php:86
     82▕         $expressionValue = $expression->getValue($grammar);
     83▕ 
     84▕         preg_match('/ST_GeomFromText\(\'(.+)\', .+(, .+)?\)/', (string) $expressionValue, $match);
     85▕ 
  ➜ 86▕         return $match[1];
     87▕     }
     88▕ 
     89▕     private function extractSridFromExpression(ExpressionContract $expression, Connection $connection): int
     90▕     {
```

The `ST_GeomFromText` specifications allows SRID not to be specified as seen in mysql and PostGis documentation:
* [Mysql](https://dev.mysql.com/doc/refman/8.4/en/gis-wkt-functions.html#function_st-geomfromtext) 
* [PostGIS](https://postgis.net/docs/ST_GeomFromText.html)

I have adjusted GeometryCast regex treat SRID as an optional parameter and return 0 when SRID is not defined.

### CODE REVIEW

**Comment by MatanYadaev** [comment]
@nickknissen Hi, thanks for the PR, I'll review it later.
What issue does it solve?

**Comment by nickknissen** [comment]
@MatanYadaev 
Currently `ST_GeomFromText('POINT(9.581389 55.724167)')` is not supported as an expression value. SRID is required.

I will add some tests.

**Comment by nickknissen** [comment]
@MatanYadaev I have updated the description with some more information about the error and added tests

File: tests/TestCase.php
Comment (MatanYadaev) [inline tests/TestCase.php:1]: I appreciate the migration-related changes, however, I'm not sure how it's better. I would appreciate if you could revert these changes.

File: src/GeometryCast.php [hunk 1]
```diff
@@ -76,23 +75,16 @@ public function set($model, string $key, $value, array $attributes): ?Expression
...
+    private function extractFromExpression(ExpressionContract $expression, Connection $connection): array
     {
         $grammar = $connection->getQueryGrammar();
         $expressionValue = $expression->getValue($grammar);
 
-        preg_match('/ST_GeomFromText\(\'.+\', (.+)(, .+)?\)/', (string) $expressionValue, $match);
+        preg_match("/ST_GeomFromText\(\s*'([^']+)'\s*(?:,\s*(\d+))?\s*(?:,\s*'([^']+)')?\s*\)/", (string) $expressionValue, $matches);
 
-        return (int) $match[1];
+        return [$matches[1], (int) ($matches[2] ?? 0), $matches[3] ?? ''];
```
Comment (MatanYadaev) [inline src/GeometryCast.php:88]: I think this would increase readability.

```suggestion
        return ['wkt' => $matches[1], 'srid' => (int) ($matches[2] ?? 0), 'options' => $matches[3] ?? ''];
```

Comment (MatanYadaev) [reply src/GeometryCast.php:88]: BTW, I don't think returning `options` is necessary as we don't use it today. I would appreciate if you could remove it.

File: tests/GeometryCastTest.php [hunk 2]
```diff
@@ -189,3 +189,13 @@
...
 });
+
+it('does not throw exception when raw expression value is being cast', function (string $expression): void {
+    $testPlace = TestPlace::factory()->create(['point' => DB::raw($expression)]);
+
+    $testPlace->point = null;
+    expect($testPlace->isDirty())->toBeTrue();
+})->with([
+    'without SRID' => "ST_GeomFromText('POINT(12.38057 55.73406)')",
+    'with SRID' => "ST_GeomFromText('POINT(12.38057 55.73406)', 4326)",
```
Comment (MatanYadaev) [inline tests/GeometryCastTest.php:200]: Can you add a case `with third parameter`? To make sure it doesn't miss it.

File: tests/GeometryCastTest.php [hunk 3]
```diff
@@ -189,3 +189,13 @@
 
     expect($testPlace->isDirty())->toBeFalse();
 });
+
+it('does not throw exception when raw expression value is being cast', function (string $expression): void {
+    $testPlace = TestPlace::factory()->create(['point' => DB::raw($expression)]);
+
+    $testPlace->point = null;
+    expect($testPlace->isDirty())->toBeTrue();
```
Comment (MatanYadaev) [inline tests/GeometryCastTest.php:197]:
```suggestion
    // Arrange
    $testPlace = TestPlace::factory()->create(['point' => new Point(0, 0)]);
    
    // Act
    $testPlace->update(['point' => DB::raw($expression)]);
    
    // Arrange
    expect($testPlace->point)->toEqual(new Point(55.73406, 12.38057));
```

**Comment by MatanYadaev** [comment]
Can you please resolve conflicts and fix CI?

**Comment by nickknissen** [comment]
> Can you please resolve conflicts and fix CI?

It should be up to date now.

**Comment by MatanYadaev** [comment]
Thanks!
